### DISCUSSION TITLE
How to access a service on the host's `lo` interface?

### DISCUSSION THREAD

**Discussion by DaanSelen** [NginxProxyManager/nginx-proxy-manager#4032] [General]
Just a quick question, I've tried the method to use host.docker.internal, but I cannot reach the service hosted on the host system on port 8443 from the docker container running NPM.

**Comment by shalafi99** [comment]
What happens if you try "curl https://host.docker.internal:8443" from within the NPM container?

![image](https://github.com/user-attachments/assets/b06e957d-c1d1-44a4-8413-cceb9e957a72)

Coincidentally, my docker host also has a service which listens on 0.0.0.0:8443, I added -k because it uses a self-signed certificate.

**Reply by DaanSelen** [reply]
I will test this, but does host.docker.internal work out of the box, just in case?

**Reply by shalafi99** [reply]
I believe it does but seems to be a Docker Desktop only feature. I searched for its reference and found [this](https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host):

_"I want to connect from a container to a service on the host
The host has a changing IP address, or none if you have no network access. We recommend that you connect to the special DNS name host.docker.internal, which resolves to the internal IP address used by the host.
You can also reach the gateway using gateway.docker.internal."_

For Docker in general, it might be needed to resort to using "--add-host" as in [link](https://docs.docker.com/reference/cli/docker/container/run/#add-host)

~~You mentioned that 8443 is bound to 'lo' which is 127.0.0.1 so I guess it's expected to work like the test I did here (against a service bound to 0.0.0.0 in my case).~~
**Edit:** _this explanation seems to not be precise enough regarding how the special DNS name host.docker.internal works, after searching a bit more and reading about kinds of issues related to failures for a container to resolve it. Better to be on the safe side and_ 😶.

**Reply by DaanSelen** [reply]
I can bind it to the docker0 interface, which might make it available to the Nginx Proxy Manager instance, I got a basic version working on IP basis.

EDIT1: 172.17.0.1 is the IP of the host, with the 16 submask.
